src/shims/fs.rs

```diff
@@ -600,6 +579,27 @@ impl FileMetadata {
             std::fs::symlink_metadata(path)
         };
 
+        FileMetadata::from_meta(ecx, metadata)
+    }
+
+    fn from_fd<'tcx, 'mir>(
+        ecx: &mut MiriEvalContext<'mir, 'tcx>,
+        fd: i32,
```

Comment:
Is there any chance that this `fd` might not coincide with the `fd` of the `FileHandler` in case the file is open?

Reply:
If either function gets called with a file descriptor number that was never returned by open, then it will return with an error. Both functions ultimately call `FileMetadata::from_fd`, which tries to look up a `FileHandle` from a `HashMap` by its fd, and the map only ever gets inserted to during calls to `open`. (Note that the file descriptor numbers seen by the interpreted program are assigned sequentially by Miri, and are not related to file descriptor numbers in the host Miri process) Does this answer your question?

Reply:
> Both functions ultimately call `FileMetadata::from_fd`, which tries to look up a `FileHandle` from a `HashMap` by its fd ... Does this answer your question?

Yes! this is what I wanted to know. LGTM